api/handlers.go

```diff
@@ -552,6 +552,14 @@ func getEvents(c *context, w http.ResponseWriter, r *http.Request) {
 	c.eventsHandler.Wait(r.RemoteAddr, until)
 }
 
+// POST /exec/{execid:.*}/start
+func postExecStart(c *context, w http.ResponseWriter, r *http.Request) {
+	if r.Header.Get("Connection") == "" {
```

Comment:
it's currently the only way I found to check is it's hijacked by the cli or not. hijack uses "Connection: Upgrade" otherwise it's nothing.